immune_subtype_clustering/prepare_data_sheet.py

```diff
@@ -0,0 +1,106 @@
+import os
+from typing import List
+
+import pandas as pd
+import synapseclient
+from synapseclient import File
+
+
+def syn_login():
+    syn = synapseclient.Synapse()
+    syn.login()
+    return syn
+
+
+def download_data_files(parent: str, syn: synapseclient.Synapse):
+    """downloads all TSV files in parent synapse folder from synapse id provided"""
+    children = syn.getChildren(parent=parent)
+    file_entity_list = []
+    for child in children:
+        if child["name"].endswith(".tsv"):
+            file = syn.get(child["id"])
+            file_entity_list.append(file)
+    return file_entity_list
+
+
+def load_data_files(file_entity_list: List[synapseclient.File]) -> List[pd.DataFrame]:
+    """loads all files in data_diectory into dataframes, creates Hugo column,
+    grabs Hugo and tpm columns, renames tppm column to patient id, appends each df to gene_df_list
+    """
+    gene_df_list = []
+    for file in file_entity_list:
+        patient_id = file.name.split("_")[0]
+        df = pd.read_csv(file.path, sep="\t")
+        df.insert(0, "Hugo", df["target_id"].str.split("|", expand=True)[5])
+        gene_df = df[["Hugo", "tpm"]]
+        gene_df = gene_df.rename(columns={"tpm": patient_id})
+        gene_df_list.append(gene_df)
```

Comment:
What are your thoughts on adding a transform function that removes the duplicated hugo symbols by taking the max value.

Reply:
I added a `transform_df` function which includes all transformations that we are applying to the component dfs in this script.